### ISSUE TITLE
Changing the theme of a subsection of a website, like an annual report

### ISSUE THREAD

**Issue by aleayr** [govCMS/GovCMS7#142]
Looking for community opinions on how to approach adding functionality to allow site builders to add new 'subsections' of a website that utilise a completely different theme to the base theme (ie, changing the theme based on path, etc).

Would people see this as a worthwhile addition?

If so. keen to explore options that don't involve modifying the settings.php file, I've had a suggestion of the themekey module as a possible solution, do people have an opinion on that?

@fiasco: Would be interested in your thoughts from a hosting platform perspective. Can we manage multiple themes currently through the 'Manage theme repository' section, or would work need to be done in this space?

@invisigoth, @jozhao: thoughts?

**Comment by fiasco** [comment]
Given that site builders have the freedom to use their own themes outside of the govCMS distro, so long as we were not adding more themes to the distro, I don't have a problem with adding this type of capability. It may open up for the govCMS theme to be used more broadly which would be a bonus. 

However, I think Context module also provides this which is already in govCMS. So I'm not sure a change is needed here, perhaps just some documentation on how to do it.

**Comment by WebProject2015** [comment]
It is possible to use Context to pass variables to the theme, it doesn't let you switch themes entirely, but you can change the entire appearance of some parts of the site.  Context is not as powerful as Themekey, and it isn't as clean since you basically end up implenting multiple themes worth of styles inside one theme.   There are also some complex Rules based solutions we've seen, but ultimately Themekey is much easier to maintain.    

In other words, Themekey is the best option in our opinion and we would definately use it.

**Comment by aleayr** [comment]
@fiasco Yes, definitely not looking to add more default themes to the distro.

Customers using the distro on their own hosting environment don't actually have an issue using multiple themes because they control the entire codebase, and could either add Themekey themselves, or utilise a settings.php switch for basic changes.

It's probably a question more so of whether the SaaS platform can support syncing multiple themes from git repos or whether it would have to be a feature request to the Site Factory team before this theme switching functionality could be added for the benefit of SaaS customers?

**Comment by pandaskii** [comment]
second @aleayr that it is not a problem for own hosting but a feature request to ACSF regarding "support 
multiple themes"

**Comment by aleayr** [comment]
This is something I'm keen to explore further after more customers have spoken to me about it recently. ThemeKey seems popular, what are thoughts on the downsides of ThemeKey itself to solve this issue?

**Comment by fiasco** [comment]
The downside of themekey is that it is new functionality that opens up govCMS to be more easily misused as a multisite instance. This lends itself to more support issues if site builders take this path. It's not a pathway I'd recommend govCMS support as it opens itself to many levels of access control issues which quite obviously is far from the intended use of themekey.

This issue is about alternate theming for reports which could be done with preprocess and alter hooks in a theme's template.php file.

**Comment by Feng-Shui** [comment]
WFIW, re the context solution, we've used [Context Reaction Theme](https://www.drupal.org/project/context_reaction_theme)

**Comment by stooit** [comment]
We've used context to build very basic "sub-sites" and just added the appropriate sub-site class to the body tag.

See https://www.communications.gov.au/ vs:
- https://www.communications.gov.au/what-we-do/bureau-communications-research (sub-site)
- https://www.communications.gov.au/what-we-do/internet/stay-smart-online (sub-site)

Obviously this isn't the same thing as applying an entirely new theme, but you can reuse existing content structures and get a fairly good result with minimal effort.

@aleayr the SaaS offering lets you have multiple themes -- just have 2 theme dirs in the root of the repo and you're good to go. We use this approach as we sometimes sub-theme Shiny as well as having the public theme
